### DISCUSSION TITLE
Vectors defined in .cc file do not appear in .vec output file (WBAN)

### DISCUSSION THREAD

**Discussion by asadalibajwaa** [omnetpp/omnetpp#999] [Q&A]
Hello everyone, I am using OMNET++ to implement a WBAN network. The network is defined in the WBAN.ned file. There are multiple sensors that I have used and they are defined by Sensor module that extend ApplicationLayerNodeBase and is defined in Sensor.ned. Now the issue is that I have defined some vectors in Sensor.cc to calculate delay, remaining energy and throughput, but they do not appear in the output .vec file. Can you please help me to guide where I went wrong? what could be the possible reason? I did not get any errors in build process. Thanks for your help.

[WBAN files(ned, cc, and ini).docx](https://github.com/omnetpp/omnetpp/files/11817237/WBAN.files.ned.cc.and.ini.docx)

**Comment by aarizaq** [comment]
You don't need to register the vector in the ned file, this is necessary if you are going to use signals to collect the statistics.
If you use the cOutVector directly in the code, you need to assign a name.
cOutVector remainingEnergyVector("remain energy");

**Reply by asadalibajwaa** [reply]
Thanks for your reply. I have tried this method too like this: 

`cOutVector remainingEnergyVector; in class Sensor
remainingEnergyVector.setName("NewEnergyVector") in initialize;
remainingEnergyVector.record(remainingEnergy); in handleMessage`

The problem still remains the same. I get different vectors in the .vec file but I do not get this one which is defined in Sensor.cc.

**Reply by asadalibajwaa** [reply]
I think the issue is different. I used cout statements in the override functions Initiailize() and handleMessage(), but these statements do not appear during simulation. This means that these override functions where I have defined everything are not invoked. The class Sensor is included though. I checked through ./exe -h classes. Any thoughts on why overridden functions may not be invoked? Thanks

**Reply by aarizaq** [reply]
I have just seen your configuration files and this can be a bit messy.
 ApplicationLayerNodeBase is a compound module, this module has many small modules and every one of them has a C++ implementation.
But you have replaced this complex module with a simple module.
  @class(Sensor);
Define_Module(Sensor);

One of two, or your module is not running in your simulation, and you can see the packet transmissions in the IDE, or your module is running, but you cannot transmit.

Your module should be implemented as an application module, not as a node. 
You can implement your module as a UDP application.

You can see an example in showcases/wireless/sensornetworks
You can see that this example uses SensorNode.ned but this doesn't use @class and doesn't replace the compound module using a simple module as you are doing using @class and Define_Module.

**Reply by asadalibajwaa** [reply]
Yeah! You are right! My module is not running in the simulation and I see the packet transmission in the Simulation IDE. If I implement it as an application module, and not as a node, would it be able to extend all the properties of SensorNode? I just want to use already existing SensorNode and then select ClusterHead on the basis of energy and distance. Then, I would define vectors that would record some values. Can you please further guide me about If I implement my module as UdpBasicApp, how would I incorporate SensorNode? Thanks
